src/main/webapp/app/exercises/programming/manage/build-plan-editor.component.ts

```diff
@@ -71,10 +73,17 @@ export class BuildPlanEditorComponent implements AfterViewInit, OnInit {
      */
     private loadBuildPlan(exerciseId: number) {
         this.isLoading = true;
-        this.buildPlanService.getBuildPlan(exerciseId).subscribe((buildPlan) => {
-            this.buildPlan = buildPlan.body!;
-            this.initEditor();
-            this.isLoading = false;
+        this.buildPlanService.getBuildPlan(exerciseId).subscribe({
+            next: (buildPlan) => {
+                this.buildPlan = buildPlan.body!;
+                this.initEditor();
+                this.isLoading = false;
+            },
+            error: () => {
```

Comment:
I think we should actually inspect the error here and only display this alert for the exact expected error (404?). There could be other error types here in exceptional cases, right?

Reply:
Only for 404 errors the specific message is shown now, for other errors the generic global error handler is called b51300e284ac96088e40aa436bedfd26629fa5f0.